File: app/src/main/java/example/homework/MusicService.java
package example.homework;

import android.app.Service;
import android.content.Intent;
import android.media.MediaPlayer;
import android.os.Binder;
import android.os.IBinder;

import java.util.ArrayList;

public class MusicService extends Service implements MediaPlayer.OnPreparedListener {

    private MediaPlayer mediaPlayer;
    private MusicBinder binder;
    private ArrayList<Item> items;
    private Item item;
    private int position;

    @Override
    public void onCreate() {
        super.onCreate();
        binder = new MusicBinder();
    }

    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {
        items = intent.getParcelableArrayListExtra("list");
        return super.onStartCommand(intent, flags, startId);
    }

    @Override
    public IBinder onBind(Intent intent) {
        return binder;
    }

    @Override
    public void onPrepared(MediaPlayer mediaPlayer) {
        mediaPlayer.start();
    }

    /**
     * Play/Pause
     */
    public boolean play() {
        if (mediaPlayer.isPlaying()) {
            mediaPlayer.pause();
            return true;
        } else {
            mediaPlayer.start();
            return false;
        }
    }

    /**
     * To Previous Soundtrack
     */
    public String previous() {
        if (mediaPlayer == null) return "Error";
        if (position == 0)
            position = items.size() - 1;
        else
            position--;
        releaseMediaPlayer();
        setItem(position);
        return item.getName();
    }

    /**
     * To Next Soundtrack
     */
    public String next() {
        if (mediaPlayer == null) return "Error";
        if (position == items.size() - 1)
            position = 0;
        else
            position++;
        releaseMediaPlayer();
        setItem(position);
        return item.getName();
    }

    /**
     * Binder
     */
    class MusicBinder extends Binder {
        MusicService getService() {
            return MusicService.this;
        }
    }

    /**
     * Getters/Setters
     */
    public void setItem(int position) {
        item = items.get(position);
        mediaPlayer = MediaPlayer.create(this, item.getId());
    }

    public Item getItem() {
        return item;
    }

    /**
     * Сleaning Up Resources
     */
    public void releaseMediaPlayer() {
        if (mediaPlayer != null) {
            try {
                mediaPlayer.release();
                mediaPlayer = null;
            } catch (Exception e) {
                e.printStackTrace();
            }
        }
    }
}

File: app/src/main/java/example/homework/MusicService.kt
package example.homework

import android.app.Service
import android.content.Intent
import android.media.MediaPlayer
import android.os.Binder
import android.os.IBinder
import java.util.*

class MusicService : Service(), MediaPlayer.OnPreparedListener {

    private var mediaPlayer: MediaPlayer? = null
    private var binder: MusicBinder? = null
    private var items: ArrayList<Item>? = null
    var item: Item? = null
        private set
    private var position: Int = 0

    override fun onCreate() {
        super.onCreate()
        binder = MusicBinder()
    }

    override fun onStartCommand(intent: Intent, flags: Int, startId: Int): Int {
        items = intent.getParcelableArrayListExtra("list")
        return super.onStartCommand(intent, flags, startId)
    }

    override fun onBind(intent: Intent): IBinder? {
        return binder
    }

    override fun onPrepared(mediaPlayer: MediaPlayer) {
        mediaPlayer.start()
    }

    /**
     * Play/Pause
     */
    fun play(): Boolean {
        if (mediaPlayer!!.isPlaying) {
            mediaPlayer!!.pause()
            return true
        } else {
            mediaPlayer!!.start()
            return false
        }
    }

    /**
     * To Previous Soundtrack
     */
    fun previous(): String? {
        if (mediaPlayer == null) return "Error"
        if (position == 0)
            position = items!!.size - 1
        else
            position--
        releaseMediaPlayer()
        setItem(position)
        return item!!.name
    }

    /**
     * To Next Soundtrack
     */
    operator fun next(): String? {
        if (mediaPlayer == null) return "Error"
        if (position == items!!.size - 1)
            position = 0
        else
            position++
        releaseMediaPlayer()
        setItem(position)
        return item!!.name
    }

    /**
     * Binder
     */
    internal inner class MusicBinder : Binder() {
        val service: MusicService
            get() = this@MusicService
    }

    /**
     * Getters/Setters
     */
    fun setItem(position: Int) {
        item = items!![position]
        mediaPlayer = MediaPlayer.create(this, item!!.id)
    }

    /**
     * Сleaning Up Resources
     */
    fun releaseMediaPlayer() {
        if (mediaPlayer != null) {
            try {
                mediaPlayer!!.release()
                mediaPlayer = null
            } catch (e: Exception) {
                e.printStackTrace()
            }
        }
    }
}
